tools/src/test/java/org/apache/kafka/tools/ListShareGroupTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools;
+
+import joptsimple.OptionException;
+
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+import org.apache.kafka.clients.admin.ListShareGroupsOptions;
+import org.apache.kafka.clients.admin.ListShareGroupsResult;
+import org.apache.kafka.clients.admin.MockAdminClient;
+import org.apache.kafka.clients.admin.ShareGroupListing;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.ShareGroupState;
+import org.apache.kafka.test.TestUtils;
+import org.apache.kafka.tools.ShareGroupsCommand.ShareGroupService;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ListShareGroupTest extends ShareGroupsCommandTest {
```

Comment:
I think this is a little confused. This is a unit test which uses a mocked admin client. It does not require a broker at all and it does not need to be parameterized for KRaft. I suggest you rename this test to `ShareGroupsCommandTest`, and then rename `ShareGroupsCommandTest` to `ShareGroupsCommandIntegrationTest`. Follow the pattern of `TopicCommandTest` and `TopicCommandIntegrationTest`.

The tests themselves look good to me, but the way that the classes are defined seems a little wrong. In fact, there's not a lot to change, but the end result would be very different.

Reply:
Oops, my bad. That makes complete sense. I have made the required changes and pushed a new commit. Thanks a lot !